### DISCUSSION TITLE
RFC: Datatables

### DISCUSSION THREAD

**Discussion by gfellerph** [swisspost/design-system#1200] [Polls]
@swimlane/ngx-datatable seems not to be really actively maintained. In order to standardise a new implementation, a survey was conducted to identify the most relevant features a datatable component should have. You can find the results below.

<img width="708" alt="Diagram listing features for a datatable component and number of answers by survey participants" src="https://user-images.githubusercontent.com/1659006/221857774-d6d37833-745c-4cf4-8768-ffae01596fcb.png">

For a new component, these features should be taken into account. Also other criteria play a role in choosing a new implementation.

| Package | Compatibility | Filtering, Sorting, Paging | Accessibility | Customizable | License and Price |
|-|-|-|-|-|-|
| [ngx-datatable](https://github.com/swimlane/ngx-datatable)             | Angular                       | ✅                          | ⛔️            | ✅            | MIT                                        |
| [PrimeNg](https://primeng.org/table)                                   | Angular                       | ✅                          | ?             | ✅            | MIT, custom styles 500.- per Major version |
| [Telerik Kendo UI Grid](https://www.telerik.com/kendo-angular-ui/grid) | Angular, VueJS, React, jQuery | ✅                          | ✅             | ✅            | From $899 per developer                    |
| [Material Design DataTable (mat-table)](https://material.angular.io/components/table/overview) | Angular | ✅ | ✅ | ✅ | MIT |
| [Smart HTML Table](https://www.htmlelements.com/demos/table/overview/) | Web Component | ✅ | ✅ | ✅ | Enterprise plan (https://www.htmlelements.com/license/) |
| [AG Grid](https://www.ag-grid.com/example/) | All frameworks | ✅  | ✅ | ✅ | Licensing is complicated: https://www.ag-grid.com/license-pricing (asked for a quote on custom licensing) |

More datatables: https://github.com/FancyGrid/awesome-grid

**Comment by pazoozooCH** [comment]
@gfellerph , a Swiss Post colleague told me about this poll / discussion (even though discussion might be a bit of an exaggeration... ;-) )
We also recently had some sorting requirements for a table and I thought about using `@swimmlane/ngx-datatable` as we've used it before in other project in Post. But the PR and issue state in Github drove me off.

I think it would be good if there was like an officially supported library at Swiss Post for the most common grid requirements. What's the status about this? Might also be helpful if that was referenced in the Design System documentation.

Is there a reason [AG Grid](https://www.ag-grid.com/) isn't on the list? It's not perfect, but I think it should at least be in the conversation given its maturity and popularity...

**Reply by gfellerph** [reply]
So far, two favorites were mentioned. For Angular projects, primeNG seems to be a good fit, for other projects AG grid can be a good fit because it's supported with any framework. For AG grid, I'm not sure how we should handle licensing, but styles could be provided by the Design System.

We're planning to involve more devs into the decicion making for this one, but we're not quite ready yet. Therefore it's still a draft.
